testing/todocheck_test.go

```diff
@@ -397,6 +399,42 @@ func TestConfigDerivedFromBasepath(t *testing.T) {
 	}
 }
 
+func TestConfigAutoDetect(t *testing.T) {
+	// Since we're unable to have a .git directory anywhere inside the repository,
+	// make sure it's there beforehand.
+	gitDir := "./scenarios/auto_detect_config/.git"
+	gitConfig := `[remote "origin"]
+	url = git@github.com:test_username/test_repo.git
+	fetch = +refs/heads/*:refs/remotes/origin/*
+[branch "master"]
+	remote = origin
+	merge = refs/heads/master
+`
+
+	if _, err := os.Stat(gitDir); os.IsNotExist(err) {
+		if err := os.Mkdir(gitDir, 0777); err != nil {
+			t.Errorf("%s", err)
+		}
+	}
+	if err := ioutil.WriteFile(gitDir+"/config", []byte(gitConfig), 0644); err != nil {
+		t.Errorf("%s", err)
+	}
```

Comment:
This is a nice way to handle the issue.

Could you extract it in the scenariobuilder as e.g. `WithGitConfig(the parameters added in a git config)` so that it can be reused across several tests?

Reply:
Done.